internal/s3select/select.go

```diff
@@ -522,27 +522,24 @@ func (s3Select *S3Select) Evaluate(w http.ResponseWriter) {
 	sendRecord := func() bool {
 		buf := bufPool.Get().(*bytes.Buffer)
 		buf.Reset()
-
+		defer bufPool.Put(buf)
```

Comment:
No. The buffer is handed over to the writer and cannot be reused.

The buffer is properly reused here: https://github.com/minio/minio/blob/d4e0f13bb3cde2db6e39c620f8e8f2bc37d99cf0/internal/s3select/message.go#L333

Reply:
O,yeah